context/app/static/js/helpers/functions.spec.js

```diff
@@ -85,3 +86,46 @@ test('generateCommaList', () => {
     'apples, bananas, oranges, and grapes',
   );
 });
+
+const validEmails = [
+  'username@gmail.com',
+  'u@hotmail.org',
+  '   username@gmail.com',
+  'username@gmail.com        ',
+  ' username@gmail.com ',
+  'username@harvard.edu',
+  'username@harvard.subdomain.edu',
+  '\nusername@harvard.subdomain.edu',
+  '\n\tusername@harvard.subdomain.edu',
+  'username@harvard.subdomain.edu\n',
+  'username@harvard.subdomain.edu\n\t',
+];
+
+const invalidEmails = [
+  '',
+  ' ',
+  'gmail.com',
+  '@gmail.com ',
+  'username',
+  'username@.com',
+  'user name@gmail.com',
+  'username @gmail.com',
+  'username@ gmail.com',
+  'username@gmail .com',
+  'username\t@gmail.com',
+  'username\n@gmail.com',
+  'username@gmail\n.com',
+  'username@gmail.\tcom',
+  'user\nname@gmail.com',
+  'user\tname@gmail.com',
+];
+
+test('isValidEmail', () => {
+  validEmails.forEach((email) => {
+    expect(isValidEmail(email)).toStrictEqual(true);
+  });
+
+  invalidEmails.forEach((email) => {
+    expect(isValidEmail(email)).toStrictEqual(false);
+  });
```

Comment:
This works, but it may be worth trying out `test.each` for similar cases in the future, e.g. https://github.com/hubmapconsortium/portal-ui/blob/6a8364803c0bb9890a453451fe591f7473c22326/context/app/static/js/components/detailPage/provenance/ProvAnalysisDetailsLink/ProvAnalysisDetailsLink.spec.js#L26

Reply:
Good to know, thanks!